src/torrent_summary.rs

```diff
@@ -146,4 +167,97 @@ impl TorrentSummary {
 
     table
   }
+
+  pub(crate) fn write_json(&self, env: &mut Env) -> Result<()> {
+    let torrent_summary_data = self.torrent_summary_data();
+    write!(
+      env.out_mut(),
+      "{}",
+      serde_json::to_string(&torrent_summary_data)?
```

Comment:
Without the `From` impl, you can do `serde_json::to_string(&torrent_summary_data).context(error::JsonSerialize)?`